src/__tests__/cli.ts

```diff
@@ -0,0 +1,501 @@
+import fs from 'fs';
+import path from 'path';
+import util from 'util';
+import { CliArguments } from '..';
+
+const mkdir = util.promisify(fs.mkdir);
+const rmdir = util.promisify(fs.rmdir);
+const writeFile = util.promisify(fs.writeFile);
+const readFile = util.promisify(fs.readFile);
+
+async function exec(
+  testProjectDir: string,
+  { init = false, flow = false, update = false }: Partial<CliArguments> = {},
+): Promise<{ exitCode: number | null; stdout: string; stderr: string }> {
+  const originalExit = process.exit;
+  const originalCwd = process.cwd();
+  const originalConsole = {
+    log: console.log,
+    warn: console.warn,
+    error: console.error,
+  };
+
+  try {
+    let exitCode: number | null = null;
+    let stdout = '';
+    let stderr = '';
+
+    const appendStdout = (...args: any[]): void => {
+      stdout += args.map((arg) => arg.toString()).join(' ');
+    };
+
+    const appendStderr = (...args: any[]): void => {
+      stderr += args.map((arg) => arg.toString()).join(' ');
+    };
+
+    console.log = appendStdout;
+    console.warn = appendStdout;
+    console.error = appendStderr;
+
+    process.exit = (code: number): never => {
+      exitCode = exitCode ?? code;
+      return undefined as never;
+    };
+
+    process.chdir(testProjectDir);
+
+    const main = require('..').main as (args: CliArguments) => Promise<void>;
+
+    await main({ init, flow, update });
+
+    return { exitCode: exitCode ?? 0, stdout, stderr };
+  } finally {
+    process.chdir(originalCwd);
+    process.exit = originalExit;
+    Object.entries(originalConsole).forEach(([key, value]) => {
+      console[key] = value;
+    });
+  }
+}
+
+async function createProject(
+  files: Array<{ name: string; content: string }>,
+  baseDir = '.',
+): Promise<string> {
+  const randomId = Math.floor(Math.random() * 1000000);
+
+  const testSpaceDir = path.join('.test-space', randomId.toString());
+
+  await mkdir(testSpaceDir, { recursive: true });
+
+  await Promise.all(
+    files.map((file) =>
+      mkdir(path.join(testSpaceDir, path.dirname(file.name)), {
+        recursive: true,
+      }),
+    ),
+  );
+
+  await Promise.all(
+    files.map((file) =>
+      writeFile(path.join(testSpaceDir, file.name), file.content),
+    ),
+  );
+
+  return path.join(testSpaceDir, baseDir);
+}
+
+describe('cli integration tests', () => {
+  const scenarios = [
```

Comment:
I wonder if [`jest-in-case`](https://github.com/atlassian/jest-in-case)` would have been better here?

Reply:
went ahead and did this